packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -11,188 +11,238 @@
-  type BoundFunctions,
-  type ByRoleOptions,
-  type GetByText,
-  GetByRole,
-} from "@testing-library/dom";
-
-import {
+  type AccessibleLabelSelector,
+  type AriaRole,
+  type FeatureSelector,
   type FeatureTestSelector,
   type TestSelector,
 } from "./featureTestSelector";
-import { odysseyTestSelectors } from "./odysseyTestSelectors";
-
-export const interpolateString = (
-  string: string,
-  values: Record<string, string | RegExp>,
-) => {
-  const interpolatedString = eval(`
-    ${Object.entries(values)
-      .map(
-        ([key, value]) =>
-          `const ${key} = ${
-            typeof value === "string" ? JSON.stringify(value) : value
-          };`,
-      )
-      .join("")}
-
-    \`${string}\`
-  `) as string;
-
-  if (/^\/*(.+)\/$/.test(interpolatedString)) {
-    return eval(interpolatedString) as RegExp;
-  }
-
-  return interpolatedString;
-};
-
-const getByQuerySelector = ({
-  canvas,
-  method,
-  options,
-  role,
-  text,
-}: {
-  canvas: BoundFunctions<typeof queries>;
-  method: "ByRole" | "ByLabelText" | "ByPlaceholderText" | "ByText";
-  options?: ByRoleOptions;
-  role?: Parameters<GetByRole>[1];
-  text?: Parameters<GetByText>[1];
-}) => {
-  if (method === "ByRole") {
-    return canvas.getByRole(
-      // TODO: These should eventually reference `query` as the function identifier.
-      role!,
-      options,
-    );
-  } else if (method === "ByLabelText") {
-    return canvas.getByLabelText(
-      // These should eventually reference `query` as the function identifier.
-      text!, // TODO: Use TypeScript `Infer` to ensure `label` is required when it's `ByLabelText`.
-      options,
-    );
-  } else if (method === "ByPlaceholderText") {
-    return canvas.getByPlaceholderText(
-      // These should eventually reference `query` as the function identifier.
-      text!, // TODO: Use TypeScript `Infer` to ensure `label` is required when it's `ByLabelText`.
-      options,
-    );
-  } else if (method === "ByText") {
-    return canvas.getByText(
-      // These should eventually reference `query` as the function identifier.
-      text!, // TODO: Use TypeScript `Infer` to ensure `description` is required when it's `ByLabelText`.
-      options,
-    );
-  }
-
-  return null;
-};
-
-export const querySelector = <TestSelectors extends FeatureTestSelector>({
-  canvas,
-  templateArgs: templateArgsProp,
-  testSelectors,
-}: {
+import { getComputedAccessibleText } from "./getComputedAccessibleText";
+import {
+  getByRoleQuerySelector,
+  getByTextQuerySelector,
+  type QueryMethod,
+} from "./getByQuerySelector";
+import { getControlledElement } from "./linkedHtmlSelectors";
+import { ElementError } from "./sanityChecks";
+
+export type QuerySelectorOptions<
+  LocalFeatureTestSelector extends FeatureTestSelector,
+> = LocalFeatureTestSelector extends TestSelector
+  ? Record<
+      keyof LocalFeatureTestSelector["selector"]["options"],
+      string | RegExp
+    >
+  : Record<string, string>;
```

Comment:
Is this an expected case? Or should this be a `never`?

Reply:
This type is actually unused now. But yes, it should be `never`.